KsApi/px/PerimeterXClientTests.swift

```diff
@@ -50,20 +50,13 @@ final class PerimeterXClientTests: XCTestCase {
       return
     }
 
-    var notification: Notification?
+    XCTAssertTrue(client.handleError(response: response, and: data!))
 
-    let token = NotificationCenter.default.addObserver(
-      forName: Notification.Name.ksr_perimeterXCaptcha,
-      object: nil,
-      queue: nil
-    ) { note in
-      notification = note
+    expectation(forNotification: Notification.Name.ksr_perimeterXCaptcha, object: nil) { note in
```

Comment:
This is a better way of testing for the notification as it allows us to also inspect it and check that it's posted on the correct thread.